### PR TITLE
Catching up PG and FISTA to R2 performance

### PR DESCRIPTION
**Pull request by rjbaraldi**
works with demo-bpdn. This PR makes it so that either can be used as a main algorithm or a subsolver, and outputs the stats struct. 
Do not merge until #34 is merged.

### CODE REVIEW

File: src/PG_alg.jl [hunk 1]
```diff
@@ -39,52 +60,78 @@ function PG(
...
-    hk = h(x⁺)
+    ∇f!(∇fk, xk)
+    fk = f(xk)
+    hk = h(xk)
 
     k+=1
-    err = norm(g-gold - (x⁺-x)/ν)
-    optimal = err < ϵ
-    tired = k ≥ maxIter
+    ξ = norm(∇fk .- ∇fkn .- (xk .- xkn) ./ ν)
```
[the remark this answers is not in the document]
Comment (rjbaraldi) [reply src/PG_alg.jl:104]: not necessarily. We could switch it back to ‖Gν‖. This is basically the norm of the subdifferential, which we use in the paper. I was more or less just being consistent there, but am not married to anything.

[the remark this answers is not in the document]
Comment (dpo) [reply src/PG_alg.jl:104]: Ok, that's fine with me. Do you agree with the two changes suggested?

[the remark this answers is not in the document]
Comment (rjbaraldi) [reply src/PG_alg.jl:104]: yep the changes make sense. I'll commit them.

**Comment by dpo** [comment]
thanks!

**Pull request merged**